### ISSUE TITLE
YMDK Wings Hotswap not getting detected under Windows and Ubuntu Linux after reboot

### ISSUE THREAD

**Issue by tedbarth** [the-via/releases#124]
My YMDK Wings Hotswap (no official support yet, compiled QMK code referenced in a [merge request](https://github.com/qmk/qmk_firmware/pull/14176)) works fine under windows with VIA 1.3.1. On Ubuntu 20.04 (Linux) it doesn't get detected. I'm unsure about what to do or where to ask, so I'm asking here for help.

```
ted@ted-pc:~$ ls -la /etc/udev/rules.d/92-viia.rules 
-rw-r--r-- 1 root root 84 Sep  1 23:41 /etc/udev/rules.d/92-viia.rules
```

```
ted@ted-pc:~$ cat /etc/udev/rules.d/92-viia.rules 
KERNEL=="hidraw*", SUBSYSTEM=="hidraw", MODE="0666", TAG+="uaccess", TAG+="udev-acl"ted@ted-pc:~$
```
 
 This last output looked weird (broken line feed?). I tried to fix the new line at the end but got asked by VIA for adding udev rules. So I tried with the following command:
```
DISABLE_SUDO_PROMPT=1 via 
```

Which didn't work out. Even calling it as root via sudo didn't help (even with original udev rule):
```
sudo DISABLE_SUDO_PROMPT=1 via --no-sandbox
```
 
```
ted@ted-pc:~$ lsusb | grep YMDK
Bus 001 Device 010: ID 594d:4975 YMDK WINGS HOTSWAP
```
 
Output for first execution without udev rules and entering sudo pw:
```
ted@ted-pc:~$ via
Beginning of file
Finish Imports
Requesting lock
promppting linux access
390cd24a6ad30bec63967065a5898337
loading app url
23:30:48.862 › APPIMAGE env is not defined, current application is not an AppImage
finished loading
```

```
ted@ted-pc:~$ hwinfo (there are much more lines with "YMDK Wings" in, would they be helpful?)
...
69: USB 00.1: 0000 Unclassified device
  [Created at usb.122]
  Unique ID: xnLL.wCFsJYTrlOA
  Parent ID: k4bc.029GFTetOb3
  SysFS ID: /devices/pci0000:00/0000:00:01.3/0000:03:00.0/usb1/1-4/1-4:1.1
  SysFS BusID: 1-4:1.1
  Hardware Class: unknown
  Model: "YMDK WINGS HOTSWAP"
  Hotplug: USB
  Vendor: usb 0x594d "YMDK"
  Device: usb 0x4975 "WINGS HOTSWAP"
  Revision: "0.01"
  Driver: "usbhid"
  Driver Modules: "usbhid"
  Speed: 12 Mbps
  Module Alias: "usb:v594Dp4975d0001dc00dsc00dp00ic03isc00ip00in01"
  Driver Info #0:
    Driver Status: usbhid is active
    Driver Activation Cmd: "modprobe usbhid"
  Config Status: cfg=new, avail=yes, need=no, active=unknown
  Attached to: #78 (Hub
...
```

EDIT: Even on windows now the keyboard stopped getting detected in VIA (I have rebooted only after a working keyboard and VIA in Windows). Recompiling and flashing QMK again didnt help. Something weird is happening…

Any idea what could be wrong? Thank you!

**Comment by alittlepeace** [comment]
What VIA config are you using?

**Comment by tedbarth** [comment]
@alittlepeace This deleted file here: https://github.com/qmk/qmk_firmware/pull/14176/files/74d01d361bbc2dae1aaaec0a72cd1f6cb61451bf#diff-e7b8046b22633c29f7a946450777f48318d54feecb875a192246a623522777d7

**Comment by alittlepeace** [comment]
Also, did you compile with the changes or without the changes?

**Comment by tedbarth** [comment]
With all your changes from your repo/merge request, nothing else.
FYI: I am switching to bare QMK because I need a way to change key background color of some keys depending on the current layer. But I will test everything and hope this feature will at someday be possible with VIA GUI (Or is this possible already: Custom code on layer switch + via?).

**Comment by alittlepeace** [comment]
That's possible with RGB Matrix, but not with RGB Light afaik

**Comment by tedbarth** [comment]
I now see, there is the [lightning layer feature](https://beta.docs.qmk.fm/using-qmk/hardware-features/lighting/feature_rgblight#lighting-layers) (this is what I was looking for) which if I see it correctly is compatible with VIA. This makes it even more interesting to have via support again.

Even after multiple reboots VIA still doesn't connect to my Wings HS again on Windows and Linux. It worked on Windows until I rebooted.
I hope this is not a hardware problem. I checked different USB ports as well.
Bare QMK is still working and I currently use my custom hard coded keymap with it, instead of VIA.

Maybe my EPROM or any other state on the chip (if there is one) is compromised somehow. Is there a way to clear the whole keyboard chip if not already done when flashing?
